spec/lib/advanced_search_builder_spec.rb

```diff
@@ -57,4 +57,47 @@ def initialize(blacklight_config)
       end
     end
   end
+
+  describe "#add_advanced_search_to_solr" do
+    let(:to_solr) { { q: 'advanced query', fq: 'inclusive facet' } }
+    let(:advanced_query) { double('advanced_query') }
+    let(:solr_params) { { q: 'basic', fq: 'original fq' } }
+    before do
+      allow(advanced_query).to receive(:to_solr).and_return(to_solr)
```

Comment:
With all this stubbing, I'm not sure what real world behavior we're modeling here? How does `QueryParser` return { q: 'advanced query' } when the blacklight_params has `q: 'basic'`?